t/t1001-mf-priority.t

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+test_description='Test multi-factor priority plugin'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+# BULK_UPDATE=${FLUX_BUILD_DIR}/t/scripts/send_fake_payloads.py
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+        flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'list all loaded plugins' '
+        flux jobtap list | grep mf_priority
+'
+
+test_expect_success 'try to submit a job when user does not exist in DB' '
+        test_must_fail flux mini submit -n1 hostname > failure.out 2>&1 &&
+        test_debug "cat failure.out" &&
+        grep "user not found in flux-accounting DB" failure.out
+'
+
+test_expect_success 'send an empty payload to make sure unpack fails' '
+cat <<-EOF >bad_payload.py
+import flux
+
+#create a JSON payload
+flux.Flux().rpc("job-manager.mf_priority.rec_update", {}).get()
+EOF
+flux python bad_payload.py
+flux dmesg | grep "failed to unpack custom_priority.trigger msg: Protocol error"
+'
+
+test_expect_success 'create fake_payload.py' '
+cat <<-EOF >fake_payload.py
+import flux
+import pwd
+import getpass
+
+username = getpass.getuser()
+userid = pwd.getpwnam(username).pw_uid
+# create a JSON payload
+data = {"userid": str(userid), "bank": "account3", "fairshare": "0.45321"}
+flux.Flux().rpc("job-manager.mf_priority.rec_update", data).get()
+data = {"userid": str(userid), "bank": "account2", "fairshare": "0.11345"}
+flux.Flux().rpc("job-manager.mf_priority.rec_update", data).get()
+EOF
+'
+
+test_expect_success 'update plugin with sample test data' '
+        flux python fake_payload.py
+'
+
+test_expect_success 'submit a job with default urgency' '
+        jobid=$(flux mini submit --setattr=system.bank=account3 -n1 hostname) &&
+        flux job wait-event -f json $jobid priority | jq '.context.priority' > job1.test &&
+        cat <<-EOF >job1.expected
+        45321
+        EOF
+        test_cmp job1.expected job1.test
+'
+
+test_expect_success 'submit a job with custom urgency' '
+        jobid=$(flux mini submit --setattr=system.bank=account3 --urgency=15 -n1 hostname) &&
+        flux job wait-event -f json $jobid priority | jq '.context.priority' > job2.test &&
+        cat <<-EOF >job2.expected
```

Comment:
```suggestion
        cat <<-EOF >job2.expected &&
```

Reply:
It looks like `--chain-lint` fails unexpectedly when appending `&&` after these lines (`cat <<-EOF > job.expected`), but doesn't fail when I leave them out. The most recently pushed of `t1001-mf-priority.t` passes with `--chain-lint` though:

```
[moussa1@docker-desktop t]$ ./t1001-mf-priority.t --chain-lint
ok 1 - load multi-factor priority plugin
ok 2 - check that mf_priority plugin is loaded
ok 3 - try to submit a job when user does not exist in DB
ok 4 - send an empty payload to make sure unpack fails
ok 5 - create fake_payload.py
ok 6 - update plugin with sample test data
ok 7 - submit a job with default urgency
ok 8 - submit a job with custom urgency
ok 9 - submit a job with urgency of 0
ok 10 - submit a job with urgency of 31
ok 11 - submit a job with other bank
ok 12 - submit a job using default bank
ok 13 - submit a job using a bank the user does not belong to
ok 14 - reject job when invalid bank format is passed in
# passed all 14 test(s)
1..14
```

Reply:
Hm, sorry about the apparent misinformation, but let me check into this.

Reply:
No worries - it could definitely be something I'm doing wrong. I was pretty confused that `--chain-lint` was complaining about the following line:

```
cat <<-EOF >bad_payload.py &&
```

which, if no `&&` is written, dumps out this error:

```
error: bug in the test script: broken &&-chain: 
cat <<-EOF >bad_payload.py
import flux
```

but not 

```
cat <<-EOF >job1.expected
```

which, if `&&` is written, dumps out this:

```
FATAL: Unexpected exit with code 2
flux-start: 0 (pid 16667) exited with rc=1
```

Reply:
Ok, the problem at least in some of these cases is that you are using 8 spaces instead of tabs.
Specifically, `<<-EOF` tells the shell to ignore TABS up to the next `<tab>EOF`. If you convert to tabs (which I think we've settled on for sharness tests) then I think you'll be able to add the `&&` to the `cat` lines.

The problem with leaving this `&&` off is that the commands before the `cat <<-EOF` could fail, but the test itself could succeed.

Reply:
Ahhh, okay. Yeah, I must have been converting tabs to spaces when I was saving the `.t` in my IDE. I'm so sorry for the confusion! The most recent push should have the `&&`'s added to the sharness tests.

Reply:
That would also probably explain why it would complain about an unexpected indent when creating the Python scripts inside the sharness tests. I should push up and squash up a fix that properly indents those sharness tests.

Reply:
yeah, the indents between the `cat <<-EOF` and the `EOF` have to be tabs. I think without that the `cat` will read all the way to the closing quote of the `test_expect_success()` -- so you aren't actually running the test! :-)

Reply:
Oh my goodness 😳, well, thank you for catching that and informing me why that happens. This latest push should have replaced the spaces with tabs to prevent the tests from not running.